py3-poetry-core.yaml

```diff
@@ -27,10 +34,45 @@ pipeline:
       tag: ${{package.version}}
       expected-commit: 710e59b879c6dd7e7f523f3945bfedd99afae723
 
-  - name: Python Build
-    uses: python/build-wheel
+subpackages:
+  - range: py-versions
+    name: py${{range.key}}-${{vars.pypi-package}}
+    description: python${{range.key}} version of ${{vars.pypi-package}}
+    dependencies:
+      provider-priority: ${{range.value}}
+      provides:
+        - py3-${{vars.pypi-package}}
+    pipeline:
+      - uses: py/pip-build-install
+        with:
+          python: python${{range.key}}
+      - uses: strip
+    test:
+      pipeline:
+        - uses: python/import
+          with:
+            python: python${{range.key}}
+            import: poetry
+
+  - name: py3-supported-${{vars.pypi-package}}
+    description: meta package providing ${{vars.pypi-package}} for supported python versions.
+    dependencies:
+      runtime:
+        - py3.10-${{vars.pypi-package}}
+        - py3.11-${{vars.pypi-package}}
+        - py3.12-${{vars.pypi-package}}
+    test:
+      pipeline:
+        - uses: python/import
+          with:
+            python: python3.10
+            import: poetry
```

Comment:
Yeah, testing the multi package makes sure that it's available when using the multi package version. This also fails if the package happens to install a bin because the meta package won't satisfy the dependencies due to the conflict.  It's helped me split the bin package out before I noticed they install a bin as a dep for another project.


The bottom one for the non-sub package ensures that only one version of python gets installed since for that package. Otherwise the test fails unable to pick an interpreter.

I would like to run the actual test more frequently though.